### DISCUSSION TITLE
Can we add a real user info to a Kyuubi Connection?

### DISCUSSION THREAD

**Discussion by gabry-lab** [apache/kyuubi#2455] [Q&A] [answered]
Hi, folks, 
Sometimes, we want to add the real `user name` and `user ip`, not principal,  to a kyuubi connection, are there such configs?
For example, we have a portal to submit SQLs by Kyuubi, there are lots of users, they can use a same principal, which is considered as a batch account. So, how we distinguish these users?

**Comment by yaooqinn** [comment]
Maybe you can use the portal principal to do the login/authentication and pass the actual user with `hive.server2.proxy.user`.

take user-password authentication as an example

```
bin/beeline -u ' jdbc:hive2://10.242.189.214:10009/;user=foo;password=bar;hive.server2.proxy.user=foobar'
Warn: Not find kyuubi environment file /Users/kentyao/incubator-kyuubi/conf/kyuubi-env.sh, using default ones...

Connecting to jdbc:hive2://10.242.189.214:10009/;user=foo;password=bar;hive.server2.proxy.user=foobar
Error: Could not open client transport with JDBC Uri: jdbc:hive2://10.242.189.214:10009/;user=foo;password=bar;hive.server2.proxy.user=foobar: Failed to validate proxy privilege of foo for foobar (state=08S01,code=0)
Beeline version 1.6.0-SNAPSHOT by Apache Kyuubi (Incubating)
```

you can see the information of all users - `Failed to validate proxy privilege of foo for foobar (state=08S01,code=0)`

**Reply by gabry-lab** [reply]
We can use kerberos to connect to our Spark cluster only, is it `hive.server2.proxy.user` useful in our case?

**Reply by yaooqinn** [reply]
yes

**Reply by gabry-lab** [reply]
How can I find this info in `Kyuubi Query Engine` tab

**Reply by yaooqinn** [reply]
not quite sure whether the user `foo` will show on ui.

**Reply by gabry-lab** [reply]
Then where it show ?

**Reply by wForget** [reply]
Is proxyUser a valid hadoop user? If not this doesn't seem to work.

**Comment by wForget** [comment, marked as answer]
The `client ip` has been displayed in the kyuubi engine tab. As for the `real username` we can use a special spark configuration to set it like `spark.custom.session.real.user`, unfortunately the session configuration is not displayed in the `kyuubi engine tab`, but for connection level engines, we can see it in spark conf.

**Reply by gabry-lab** [reply]
only client ip, it is not enough. For example, a user can connect to a spark cluster by a bastion, where the client ip is same with other users.

**Reply by wForget** [reply]
> only client ip, it is not enough. For example, a user can connect to a spark cluster by a bastion, where the client ip is same with other users.

We can still add a spark configuration to set the `real ip` like: `spark.custom.session.real.ip`.

**Reply by gabry-lab** [reply]
spark.custom.session.real.ip will be shown in `kyuubi engine tab`?

**Reply by wForget** [reply]
It won't be displayed at present, but we can implement it. @yaooqinn WDYT?

**Reply by gabry-lab** [reply]
My suggest is to add a proxy user, not only a custom real ip, and let it shown in the `kyuubi engine tab`

**Reply by yaooqinn** [reply]
the custom ip is not related to kyuubi, the client side can pass it and the spark system will respect it.

> It won't be displayed at present, but we can implement it. @yaooqinn WDYT?

**Reply by yaooqinn** [reply]
> My suggest is to add a proxy user, not only a custom real ip, and let it shown in the `kyuubi engine tab`

the proxy user is well shown on the tab.

**Reply by wForget** [reply]
> My suggest is to add a proxy user, not only a custom real ip, and let it shown in the `kyuubi engine tab`

If this proxy user is a valid hadoop user, you can use `hive.server2.proxy.user` configuration as yaooqinn says, this user will be shown in kyuubi engine tab, if not, we should not show it.

**Reply by wForget** [reply]
> the custom ip is not related to kyuubi, the client side can pass it and the spark system will respect it.

I mean we can show all configuration of session in `Kyuubi Engine Session Tab`.

**Reply by gabry-lab** [reply]
> > the custom ip is not related to kyuubi, the client side can pass it and the spark system will respect it.
> 
> I mean we can show all configuration of session in `Kyuubi Engine Session Tab`.

Yes, totally agree, we can show all of the session level confs in `Kyuubi Query Engine Session` page

**Reply by gabry-lab** [reply]
> 

But I can't find it in `Kyuubi Query Engine` tab

**Reply by gabry-lab** [reply]
> > the custom ip is not related to kyuubi, the client side can pass it and the spark system will respect it.
> 
> I mean we can show all configuration of session in `Kyuubi Engine Session Tab`.

proposal here https://github.com/apache/incubator-kyuubi/issues/2484
